pandas/_libs/tslib.pyx

```diff
@@ -313,11 +321,9 @@ def array_with_unit_to_datetime(
                 result = (iresult * mult).astype("M8[ns]")
 
             elif values.dtype.kind == "f":
-                fresult = (values * mult).astype("f8")
-                fresult[mask] = 0
```

Comment:
Just for my understanding, is this just a refactor (reuse `fvalues`, as it was already calculated above)?